src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -203,29 +204,52 @@ private function createRelationsForItem($item, $formattedData)
      */
     private function getRelationDataFromFormData($data)
     {
-        $relationFields = $this->getRelationFields();
+        $relation_fields = $this->getRelationFields();
 
         $relationData = [];
-        foreach ($relationFields as $relationField) {
-            $attributeKey = $relationField['name'];
-            if (array_key_exists($attributeKey, $data) && empty($relationField['pivot'])) {
-                $key = implode('.relations.', explode('.', $relationField['entity']));
+        foreach ($relation_fields as $relation_field) {
+            $relation_field['name'] = $relation_field['entity'];
+            $attributeKey = $relation_field['name'];
+
+            if (! is_null(Arr::get($data, $attributeKey)) && empty($relation_field['pivot'])) {
+                $key = implode('.relations.', explode('.', $this->getOnlyRelationEntity($relation_field)));
                 $fieldData = Arr::get($relationData, 'relations.'.$key, []);
 
                 if (! array_key_exists('model', $fieldData)) {
-                    $fieldData['model'] = $relationField['model'];
+                    $fieldData['model'] = $relation_field['model'];
                 }
 
                 if (! array_key_exists('parent', $fieldData)) {
-                    $fieldData['parent'] = $this->getRelationModel($relationField['entity'], -1);
+                    $fieldData['parent'] = $this->getRelationModel($relation_field['entity'], -1);
                 }
-
-                $fieldData['values'][$attributeKey] = $data[$attributeKey];
+                $relatedAttribute = Arr::last(explode('.', $attributeKey));
+                $fieldData['values'][$relatedAttribute] = Arr::get($data, $attributeKey);
 
                 Arr::set($relationData, 'relations.'.$key, $fieldData);
             }
         }
 
         return $relationData;
     }
+
+    public function getOnlyRelationEntity($relation_field)
+    {
+        $entity_array = explode('.', $relation_field['entity']);
```

Comment:
What? ```entity``` can have dot notation too? Since when? I thought only ```name``` can use the dot notation...

Reply:
Indeed it can. Look at this scenario:

Current Model, HasOne Address and that Address has a BelongsTo Coutry. Entity in this scenario would be `address.country`.